dockerfiles/init/modules/codenvy/templates/machine.properties.erb

```diff
@@ -64,8 +64,8 @@ che.docker.always_pull_image=true
 che.docker.api=1.20
 
 # https support
-machine.proxy_wsagent_server_url_template=<%= scope.lookupvar('codenvy::host_protocol') %>://<%= scope.lookupvar('codenvy::host_url') %>/%3$s_%2$s/%4$s
-machine.proxy_terminal_server_url_template=<%= scope.lookupvar('codenvy::host_protocol') %>://<%= scope.lookupvar('codenvy::host_url') %>/%3$s_%2$s/%4$s
+machine.proxy_wsagent_server_url_template=<%= scope.lookupvar('codenvy::host_protocol') %>://%5$s/%3$s_%2$s/%4$s
```

Comment:
it seems you're asking me to do your job ?
https://github.com/codenvy/deployment/commit/16e7d0f0916df8c132971d15d9755da51f23f264

I will provide them but you may follow your tips as well.